### PR TITLE
update status on emms system io errors

### PR DESCRIPTION
**Pull request by michelpmcdonald**
## Summary

Occasionally EMMS API hits some kind of temporary internal error when submitting documents to them:
"Upstream status: System.IO.IOException: The process cannot access the file '\\Sftpemms\d$\SFTPRoot\GCIOEM\prod\Inbound\EMMS-GCIO-19777445.ack' because it is being used by another process. at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath) at "

This error seems to self resolve without US resubmitting , but when we get this error back from our upload call to them, we end up marking the submission with a status of error.  However, the submission itself is fine.  We hit this 28 times, and in all cases the submissions on the EMMS side of things have a status of complete.

The real issue here is that after we mark the submission with an error status, we never update the status from EMMS.

To resolve, when we hit this error, UploadStatusBatch, the job that we use to update our tracking record with the latest EMMS status now updates records with this specific error and the SlackStatusNotifier will start to report when the error status is around 80 minutes old.

Team banana-peels owns this code 
  

## Related issue(s)
[API-36242
](https://jira.devops.va.gov/browse/API-36242)

Used Unit tests to reproduce issue and confirm a status update is scheduled

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
Benefits Intake upload submissions to EMMS\Central Mail when hitting a rare error.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.

## Requested Feedback

Since we don't really have an idea how long it takes for EMMS to self resolve these errored submissions and start returning non-error statuses, at this point, going to start reporting after 80 minutes and continue reporting for 30 days.  We might have to revisit these timings once we see it operate in prod and start to get a better handle on timings

### CODE REVIEW

**Review by ericboehs — DISMISSED** [review]

**Review by mathisto — APPROVED** [review]
Barring the two typos Kristen pointed out, this looks great.

File: modules/vba_documents/app/sidekiq/vba_documents/slack_status_notifier.rb [hunk 1]
```diff
@@ -30,6 +30,20 @@ class SlackStatusNotifier
...
     UPLOADED_AGE_THRESHOLD_MINUTES = 100
 
+    # After hitting an Internal EMMS error, delay reporting until the status updater has a chance
+    # to refresh the latest status since most of the time these self resolve in EMMS
+    EMMS_INTERNAL_ERROR_THRESHOLD_MINUTES = 80
+
+    EMMS_SYSTEMIO_ERROR_REPORT =
+      'Benefits Intake Submissions are in error status with an EMMS System.IO error.' \
+      ' Typically, these self resolve without our intervention and susbequent calls for the status of ' \
+      'these submission return a non-error status, but these appear stuck.  Notify EPPS API team for help'
```
[the remark this answers is not in the document]
Comment (michelpmcdonald) [reply modules/vba_documents/app/sidekiq/vba_documents/slack_status_notifier.rb:40]: aww that was a good catch

File: modules/vba_documents/app/sidekiq/vba_documents/slack_status_notifier.rb [hunk 2]
```diff
@@ -30,6 +30,20 @@ class SlackStatusNotifier
...
+
+    EMMS_DUP_CONFIRM_NUMBER_ERROR =
+      'Benefits Intake Submissions are in error status with an EMMS ConfirmationNumber has already been ' \
+      'submitted error. Typically, these self resolve without our intervention and susbequent calls for the ' \
+      'status of these submission return a non-error status, but these appear stuck.  Notify EPPS API team for help'
```
[the remark this answers is not in the document]
Comment (michelpmcdonald) [reply modules/vba_documents/app/sidekiq/vba_documents/slack_status_notifier.rb:45]: copy paste

File: modules/vba_documents/spec/models/upload_submission_spec.rb [hunk 3]
```diff
@@ -156,6 +156,17 @@ def status_complete_packets(*reasons)
...
 
+  describe 'clear_old_error' do
+    it 'clears error fields when moving out of error status' do
+      upload = FactoryBot.create(:upload_submission, status: 'error', code: 'MAV505', detail: 'Mav Error')
+      upload.status = 'received'
+      upload.save!
+      expect(upload.detail).to be_nil
+      expect(upload.code).to be_nil
+      expect(upload.status).to eq('received')
+    end
```
[the remark this answers is not in the document]
Comment (michelpmcdonald) [reply modules/vba_documents/spec/models/upload_submission_spec.rb:167]: added

[the remark this answers is not in the document]
Comment (kristen-brown) [reply modules/vba_documents/spec/models/upload_submission_spec.rb:167]: Looks great, thanks!

**Review by kristen-brown — APPROVED** [review]

**Pull request merged**